test/unit/tracks/text-track-display.test.js

```diff
@@ -183,6 +183,50 @@ if (!Html5.supportsNativeTextTracks()) {
     player.dispose();
   });
 
+  QUnit.test("don't select user langauge if it is an empty string", function(assert) {
+    const player = TestHelpers.makePlayer();
+    const track1 = {
+      kind: 'captions',
+      label: 'English',
+      language: 'en',
+      src: 'en.vtt'
+    };
+    const track2 = {
+      kind: 'captions',
+      label: 'Spanish',
+      language: 'es',
+      src: 'es.vtt'
+    };
+    const track3 = {
+      kind: 'metadata',
+      label: 'segment-metadata'
+    };
+
+    player.src({type: 'video/mp4', src: 'http://google.com'});
+    // manualCleanUp = true by default
+    const englishTrack = player.addRemoteTextTrack(track1, true).track;
+    const spanishTrack = player.addRemoteTextTrack(track2, true).track;
+    const metadataTrack = player.addRemoteTextTrack(track3, true).track;
+
+    // Force empty string ('') as "user-selected" track
+    player.cache_.selectedLanguage = { enabled: true, language: '', kind: 'captions' };
+    this.clock.tick(1);
+
+    assert.equal(spanishTrack.mode, 'disabled', 'Spanish captions should be disabled');
+    assert.equal(englishTrack.mode, 'disabled', 'English captions should be disabled');
+    assert.notEqual(metadataTrack.mode, 'showing', 'Metadata track should not be showing');
```

Comment:
Is there a reason you're opting for `notEqual`'ing `showing` instead of `equal`'ing `hidden`? Like Safari's native HLS support or something?

Reply:
Yeah, I figured it'll be safer to just make sure it isn't showing since it doesn't really matter what this is set to for this test